spec.html

```diff
@@ -42484,35 +42468,35 @@ <h1>Changes to EvalDeclarationInstantiation</h1>
-                  1. Let _thisEnvRec_ be _thisLex_'s EnvironmentRecord.
-                  1. If _thisEnvRec_ is not an object Environment Record, then
-                    1. If _thisEnvRec_.HasBinding(_F_) is *true*, then
+                  1. Let _thisScope_ be _thisLex_.[[ScopeRecord]].
+                  1. If _thisScope_ is not an object Scope Record, then
+                    1. If _thisScope_.HasBinding(_F_) is *true*, then
                       1. Let _bindingExists_ be *true*.
-                  1. Set _thisLex_ to _thisLex_'s outer environment reference.
-                1. If _bindingExists_ is *false* and _varEnvRec_ is a global Environment Record, then
-                  1. If _varEnvRec_.HasLexicalDeclaration(_F_) is *false*, then
-                    1. Let _fnDefinable_ be ? _varEnvRec_.CanDeclareGlobalVar(_F_).
+                  1. Set _thisLex_ to _thisLex_.[[OuterEnv]].
+                1. If _bindingExists_ is *false* and _varScope_ is a global Scope Record, then
+                  1. If _varScope_.HasLexicalDeclaration(_F_) is *false*, then
+                    1. Let _fnDefinable_ be ? _varScope_.CanDeclareGlobalVar(_F_).
                   1. Else,
                     1. Let _fnDefinable_ be *false*.
                 1. Else,
                   1. Let _fnDefinable_ be *true*.
                 1. If _bindingExists_ is *false* and _fnDefinable_ is *true*, then
                   1. If _declaredFunctionOrVarNames_ does not contain _F_, then
-                    1. If _varEnvRec_ is a global Environment Record, then
-                      1. Perform ? _varEnvRec_.CreateGlobalVarBinding(_F_, *true*).
+                    1. If _varScope_ is a global Scope Record, then
+                      1. Perform ? _varScope_.CreateGlobalVarBinding(_F_, *true*).
                     1. Else,
-                      1. Let _bindingExists_ be _varEnvRec_.HasBinding(_F_).
+                      1. Let _bindingExists_ be _varScope_.HasBinding(_F_).
                       1. If _bindingExists_ is *false*, then
-                        1. Perform ! _varEnvRec_.CreateMutableBinding(_F_, *true*).
-                        1. Perform ! _varEnvRec_.InitializeBinding(_F_, *undefined*).
+                        1. Perform ! _varScope_.CreateMutableBinding(_F_, *true*).
+                        1. Perform ! _varScope_.InitializeBinding(_F_, *undefined*).
                     1. Append _F_ to _declaredFunctionOrVarNames_.
                   1. When the |FunctionDeclaration| _f_ is evaluated, perform the following steps in place of the |FunctionDeclaration| Evaluation algorithm provided in <emu-xref href="#sec-function-definitions-runtime-semantics-evaluation"></emu-xref>:
                     1. Let _genv_ be the running execution context's VariableEnvironment.
-                    1. Let _genvRec_ be _genv_'s EnvironmentRecord.
+                    1. Let _gScope_ be _genv_.[[ScopeRecord]].
```

Comment:
I know this is supposed to be a fairly mechanical renaming, but I don't really like how the names used in this one algorithm stand out. Can we make them slightly more descriptive?

Reply:
Yeah, the names were kind of odd to begin with, and the mechanical renaming didn't improve things. I'd suggest:
- `_genv_` -> `_evalVarEnv_`
- `_benv_` -> `_evalLexEnv_`
- `_gScope_` -> `_evalVarScope_`
- `_bScope_` -> `_evalLexScope_`

(And similarly in the previous clause, for consistency.)

Reply:
Go for it!